### ISSUE TITLE
liberty-maven-plugin:3.0.x is not compatible with eclipse liberty integration plugin

### ISSUE THREAD

**Issue by andre-doherty** [OpenLiberty/open-liberty-tools#348]
hello,

It seems that the changes in the liberty-maven-plugin introduced in the new version 3.0.0 break the compatibility with the liberty eclipse integration plugin.
The goal install-apps which is hardcoded in the eclipse plugin has been removed in the maven-plugin. 
Therefore integration of liberty with the ide is broken.
Is there a workaround ? 
Change the setting (install-apps to deploy) in the maven task doesnt help as the task is automatically updated when the projet is re-configured in eclipse.

Thank you

André

**Comment by vp642** [comment]
I ran into the same issue and wondering if it is going to be resolved soon or at least get some workaround.

Cheers,
Vlad

**Comment by c-koell** [comment]
The same here ...

**Comment by theshovon** [comment]
Hello, 

I am the new owner of that plugin and gaining the required knowledge to find a solution/workaround. I will update you once I find a solution.

Best,
Shovon Saha.

**Comment by c-koell** [comment]
Any news here ?

**Comment by theshovon** [comment]
The call to install-apps needs to be changed to use deploy. There is no workaround. It seems like the Eclipse liberty integration plugin needs to be updated to work with the 3.x release.

**Comment by c-koell** [comment]
Maybe it is possible to check which maven plugin is used ?
It is really annoying if you use the new maven plugin and eclipse is creating a Run-Configuration which does not work ...

Please fix this bug !

**Comment by theshovon** [comment]
I have redirected the issue to the liberty team. I am guessing it is not an OLT/WDT issue. I will update if I get any further updates.

**Comment by cherylking** [comment]
@theshovon Opening an issue of the liberty maven plugin is not going to solve this issue. We made conscious breaking changes (reflected with the major version change and the release notes). Please update your tools that have hard coded calls to goals and parameters that no longer exist. You will need to check which version of the plugin is configured, and call the appropriate goals.

**Comment by andre-doherty** [comment]
Hello @theshovon,
Is there any update regarding this issue ? 
Is there a premilinary patch we can try ? 

The issue is really a pain as it breaks the integration of liberty in eclipse ....

Thank you for your feedback

André

**Comment by josephca** [comment]
@andre-doherty 
We got a fix for this and the fix will be included in the next release (March 2020). That being said, we will upload an interim build with the fix so that you can try out meanwhile. Would this work for you?

**Comment by cherylking** [comment]
Please note that liberty-gradle-plugin 3.0 release just came out last week and also has similar breaking changes. See the release notes here: https://github.com/OpenLiberty/ci.gradle/releases/tag/liberty-gradle-plugin-3.0

**Comment by andre-doherty** [comment]
> @andre-doherty
> We got a fix for this and the fix will be included in the next release (March 2020). That being said, we will upload an interim build with the fix so that you can try out meanwhile. Would this work for you?

Hi @josephca, 

Excellent news ! 
I would be more than happy to try that one out, please let me know when it's available

regards 
André

**Comment by josephca** [comment]
@andre-doherty 
I just uploaded a test build here: https://public.dhe.ibm.com/ibmdl/export/pub/software/openliberty/tools/test/348/openlibertytools-20.0.0.3.v2020-02-03_1504.zip
Thanks for your patience.

**Comment by andre-doherty** [comment]
hello @josephca 

Thank you, i have uninstalled the previous stable version i am using (WebSphere Application Server Liberty Tools	19.0.1200.v20191213_1001).

And installed this version in my eclipse ide.

At first it doesnt seem to work properly with my setup.
The liberty runtime is not detected so it doesnt go very far. (even with reimporting from scratch (no eclipse descriptor) in a new workspace).

Would you have a sample that you have tested your build against ? 

Regards

André

**Comment by rajivnathan** [comment]
@andre-doherty I saw the same issue with the linked test build so it's not just you.

@cherylking I tried a fix that uses `deploy` instead of `install-apps` but looks like that causes problems when updating a non-loose web app.

The server output shows
```
[AUDIT   ] CWWKE0001I: The server guideServer has been launched.
[AUDIT   ] CWWKE0100I: This product is licensed for development, and limited production use. The full license terms can be viewed here: https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/license/base_ilan/ilan/20.0.0.2/lafiles/en.html
[AUDIT   ] CWWKZ0058I: Monitoring dropins for applications.
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/ServletSample/
[AUDIT   ] CWWKZ0001I: Application ServletSample started in 0.257 seconds.
[AUDIT   ] CWWKF0012I: The server installed the following features: [localConnector-1.0].
[AUDIT   ] CWWKF0011I: The guideServer server is ready to run a smarter planet. The guideServer server started in 1.569 seconds.
[ERROR   ] CWWKZ0013E: It is not possible to start two applications called ServletSample.
[AUDIT   ] CWWKG0016I: Starting server configuration update.
[AUDIT   ] CWWKG0018I: The server configuration was not updated. No functional changes were detected.
```

The problem appears to be with the line `[ERROR   ] CWWKZ0013E: It is not possible to start two applications called ServletSample.`

Could this be a bug with the deploy goal or is there another goal we can use?

Note: No issues with <looseApplication>true</looseApplication>

**Comment by cherylking** [comment]
@rajivnathan I will need more information, but that error usually occurs when the application is not found in the server.xml (webApplication or application element - problem resolving the location or app name) and it generates something in the dropins folder instead. If the application is also in the apps folder, then both apps get started with the same name.  So please provide the pom.xml, server.xml and general description of the scenario.

**Comment by rajivnathan** [comment]
Scratch my last comment. I saw that problem with versions `3.0.1` of `liberty-maven-plugin` and `liberty-maven-app-parent` but I tried version `3.2` for both and I don't see that problem anymore.

So think we can use `deploy` for the fix for versions `3.x.x` and above. For versions prior to that it doesn't look like `deploy` works so we'll keep using `install-apps` for them.

**Comment by rajivnathan** [comment]
I put up a potential fix in https://github.com/OpenLiberty/open-liberty-tools/pull/382 with instructions on how to test. I don't have access to upload an update site for testing but maybe @josephca can help with that if we need it.

**Comment by josephca** [comment]
@andre-doherty @c-koell Please test it with a nightly driver ( https://public.dhe.ibm.com/ibmdl/export/pub/software/openliberty/tools/nightly/2020-03-02_1855/openlibertytools-20.0.0.3.v2020-03-02_1855.zip ) which contains the fix from @rajivnathan and please let us know if you see any issue. Again, thanks for your patience.

**Comment by andre-doherty** [comment]
hi @josephca, 

I have tested the fix over the last 2 days, looks like we can close the issue now, as everythink is fine now ! 
Thanks a lot for the job done.

André

**Comment by c-koell** [comment]
I can only join here .. tested it and it works fine .. thanks a lot for the fix !

**Comment by josephca** [comment]
@andre-doherty  @c-koell Thanks a lot for your quick test results. The fix is currently included in the nightly build and also will be included in the 20.0.0.3 release (ETA: Mar 13, 2020). 
@rajivnathan - thanks for the fix! 🥇
